### PR TITLE
Generate and read benchmark data from files

### PR DESCRIPTION
**Pull request by iliailia**
This adds a binary that can generate plain vector and graph stores, whose on-the-fly generation takes most of the running time in benchmarks. Reading this data from files takes only 4 vs 20s for 10,000 irises and 30 vs 378s for 100,000 irises.

### CODE REVIEW

**Review by bgillesp — COMMENTED** [review]
Great improvement for dev experience!  Thanks for putting this together.  Summary of comments/suggestions:
- Right now the new binary generates two redundant representations of the list of iris codes (`store.ndjson` and `irises.ndjson`), so probably we should use just one (and update deserialization appropriately)
- I gave a suggestion for making the generation binary work a bit more portably in terms of filesystem interaction
- It would be nice if the benchmark gave an informative error message and return gracefully when it doesn't find the expected files rather than panicking
- Update the `.gitignore` file to ignore the new `data` directory since these are probably transient files that we don't want to store in version control and have to keep up to date actively

File: iris-mpc-cpu/bin/generate_benchmark_data.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+use aes_prng::AesRng;
+use iris_mpc_cpu::{
+    hawkers::plaintext_store::PlaintextStore,
+    py_bindings::{io::write_bin, plaintext_store::to_ndjson_file},
+};
+use rand::SeedableRng;
+use std::error::Error;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
```
[the remark this answers is not in the document]
Comment (iliailia) [reply iris-mpc-cpu/bin/generate_benchmark_data.rs:10]: Thanks, implemented

File: iris-mpc-cpu/benches/hnsw.rs [hunk 2]
```diff
@@ -162,21 +162,33 @@ fn bench_gr_primitives(c: &mut Criterion) {
     });
 }
 
+/// To run this benchmark, you need to generate the data first by running the
+/// following commands:
+///
+/// cargo build --release
+/// ./target/release/generate_benchmark_data
```
[the remark this answers is not in the document]
Comment (iliailia) [reply iris-mpc-cpu/benches/hnsw.rs:169]: Yep, added

File: iris-mpc-cpu/benches/hnsw.rs [hunk 3]
```diff
@@ -162,21 +162,33 @@ fn bench_gr_primitives(c: &mut Criterion) {
...
+            LocalNetAby3NgStoreProtocol::lazy_setup_from_files_with_grpc(
+                "./data/irises.ndjson",
+                "./data/store.ndjson",
+                &format!("./data/graph_{}.ndjson", database_size),
+                &mut rng,
+                database_size,
+                false,
+            )
+            .await
+            .unwrap()
```
[the remark this answers is not in the document]
Comment (iliailia) [reply iris-mpc-cpu/benches/hnsw.rs:191]: Good idea, added.

File: iris-mpc-cpu/src/hawkers/galois_store.rs [hunk 4]
```diff
@@ -355,6 +356,85 @@ impl LocalNetAby3NgStoreProtocol {
...
+            .into_iter()
+            .take(database_size)
+            .collect();
+        let plaintext_vector_store = from_ndjson_file(plainstore_file, Some(database_size))
+            .map_err(|e| eyre::eyre!("Cannot find store: {e}"))?;
+        let plaintext_graph_store: GraphMem<PlaintextStore> =
+            read_bin(plaingraph_file).map_err(|e| eyre::eyre!("Cannot find graph: {e}"))?;
+
+        let protocol_stores =
+            setup_local_aby3_players_with_preloaded_db(rng, cleartext_database, network_t).await?;
```
[the remark this answers is not in the document]
Comment (iliailia) [reply iris-mpc-cpu/src/hawkers/galois_store.rs:391]: Done except for `PlaintextPoint` and `PlaintextIris` refactoring. It's better to deal with it in the following PR.

File: .gitignore [hunk 5]
```diff
@@ -5,5 +5,6 @@
 .env
 .envrc
 *.iml
+/iris-mpc-cpu/data
```
Comment (bgillesp) [inline .gitignore:8]: Just a nit: can we move this into a new `.gitignore` file with entry `/data` in the `iris-mpc-cpu` subdirectory?  It doesn't seem like a specification that is relevant to the overall workspace.

**Review by bgillesp — APPROVED** [review]
Looking good, thanks!  Small nit in comments about the `.gitignore` file, otherwise good to go!

**Pull request merged**